components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/WebsocketInboundHandler.java

```diff
@@ -259,20 +263,19 @@ private boolean validateOAuthHeader(FullHttpRequest req) throws APISecurityExcep
 
                 //Initial guess of a JWT token using the presence of a DOT.
 
-                SignedJWT signedJWT = null;
+                SignedJWTInfo signedJWTInfo = null;
                 String keyManager = null;
                 if (StringUtils.isNotEmpty(apiKey) && apiKey.contains(APIConstants.DOT)) {
                     try {
                         // Check if the header part is decoded
-                        Base64.getUrlDecoder().decode(apiKey.split("\\.")[0]);
                         if (StringUtils.countMatches(apiKey, APIConstants.DOT) != 2) {
```

Comment:
we don't need this as well right? I think that will be handled in SignedJWT.parse()

Reply:
its already deleted